### ISSUE TITLE
Do you have apk or release available?

### ISSUE THREAD

**Issue by pmoralesp** [starryalley/CSC_BLE_Bridge#12]
Thank you for your work. Do you have apk available or need to compile by myself? 

I am linux user but have a Window partition just for zwift. Some days ago I tried Zwift on Lutris and works flawless but my only concern I can't bridge my Garmin HRM-TRI (Ant+) to zwfit (others sensors has bluetooth so works fine throug zwift companion). I would like to test my setup and give some feedback/PR.

**Comment by larryb84** [comment]
I would also like to volunteer to beta test if you have an APK available? Also have a legacy Garmin ANT+ HR Monitor and a Galaxy Note 9 that supports ANT+. Like you, would like to get the HR Monitor working on Zwift without the expense of a new HR monitor or bridging device such as the Cable. Thanks for your awesome work trying to solve this. Larry

**Comment by starryalley** [comment]
Thanks @pmoralesp . I'm also a linux user but I never know Zwift on Lutris! Very interesting to know it!

@pmoralesp @larryb84  I've just included a release apk here. Feel free to try it out:
https://github.com/starryalley/CSC_BLE_Bridge/tree/master/app/release

I'd like to highlight the great work from @pinkemma who added heart rate sensor support into this app. My bluetooth heart rate sensor broke a while ago so I've been using this app to bridge my Garmin HRM-Run (Ant+ only) when zwifting!

**Comment by larryb84** [comment]
Thanks @starryalley, @pinkemma 

I just downloaded the APK to my Galaxy Note 9 and successfully managed to bridge my Garmin HRM2 Ant+ to my iPad running zwift 👍👍

The connection was very stable with no latency issues so great work. I did have to manually search and pair the iPad to my Note 9 on Bluetooth first, which took me a few tries, but after a couple of goes it showed up on the Zwift pairing screen.

Is the manual pairing necessary or will the app cause the BLE signal to broadcast automatically?

Thanks again for making this awesome solution 👍

Larry

**Comment by starryalley** [comment]
Thanks for confirming this. For now it is necessary to manually pair each time because the BLE device name changes every time this app starts. See https://github.com/starryalley/CSC_BLE_Bridge/issues/13

**Comment by IzzySoft** [comment]
May I chime in to (hopefully) make your live a little easier? Thanks to the APK now being available in the repo here, I've added it to my F-Droid repo – it should [show up here](https://apt.izzysoft.de/fdroid/index/apk/idv.markkuo.cscblebridge) with the next regular sync in about 18.5h (shortly before 8pm CET/UTC+1). You all can then use the [F-Droid](https://f-droid.org/) client and add my repo to have easy access (not only) to this app, and automatically get updates etc.

The latter requires a little update here, though: although being in development for a while now, it still identifies as "very first version":

    package: name='idv.markkuo.cscblebridge' versionCode='1' versionName='1.0' compileSdkVersion='30' compileSdkVersionCodename='11'

Let's ignore the last 2 values, and the first one is perfect of course. `versionName` is what is presented to the user and should help them to tell you which version they have an issue with (or want an improvement for), so you probably want to update that with each release. The crucial value however is `versionCode` – as that is what Android uses to tell versions apart. Given 2 packages A and B with the same packageName, and A being installed on the user's device:

* `A < B`: "Dear user, here's a new update, please install!!!"
* `A = B`: "Please move on, there is nothing to see here. Oh, you found this? You really want B? OK, if you insist…" (i.e.: no notification, but the user can "force-install" it)
* `A > B`: "This is a downgrade. No, you cannot install/update to it." (i.e. Android refuses)

So in order for users to be notified, always increase `versionCode` whenever you make a new release :smile:

While my updater can pick the APK from inside your repo, it's not the ideal place to put it: Each contributor, when forking/cloning, would have to take the entire history of that binary along. Better start [tagging releases](https://github.com/starryalley/CSC_BLE_Bridge/releases) and attach the APK there.

Oh: and if you wish, I've got [a nice badge here](https://gitlab.com/IzzyOnDroid/repo/#if-my-app-is-available-here-do-you-have-a-badge-i-can-use-to-link-to-it) to put into your Readme and point to above link in my repo :smiley:

[<img src="https://gitlab.com/IzzyOnDroid/repo/-/raw/master/assets/IzzyOnDroid.png" width="250">](https://apt.izzysoft.de/fdroid/index/apk/idv.markkuo.cscblebridge)

```markdown
[<img src="https://gitlab.com/IzzyOnDroid/repo/-/raw/master/assets/IzzyOnDroid.png" width="250">](https://apt.izzysoft.de/fdroid/index/apk/idv.markkuo.cscblebridge)
```

PS: Should you want your app listed with the F-Droid main repo, which is a bit more prestiguous, I can help you with that, too. Disclosure: I'm one of the F-Droid maintainers.

**Comment by starryalley** [comment]
Thanks @IzzySoft . I've added a release tag and [released v1.1](https://github.com/starryalley/CSC_BLE_Bridge/releases) there in the repo. I've also added the badge in the README, removed the apk in the repo.

**Comment by IzzySoft** [comment]
Thanks, wonderful! I've updated the metadata here and manually triggered the updater to see if it's found – worked out perfectly, update should show up in about 2.5h with the next sync. And thanks for adding the badge!
